src/main/groovy/org/kordamp/gradle/plugin/jandex/tasks/JandexTask.groovy

```diff
@@ -59,96 +50,143 @@ import javax.inject.Inject
  */
 @CompileStatic
 class JandexTask extends DefaultTask {
-    private final BooleanState processDefaultFileSet
-    private final BooleanState includeInJar
-    private final StringState indexName
-    private final IntegerState indexVersion
+    private final Property<Boolean> processDefaultFileSet
+    private final Property<Boolean> includeInJar
+    private final Property<String> indexName
+    private final Property<Integer> indexVersion
     private final WorkerExecutor workerExecutor
 
     @Classpath
-    Configuration classpath
+    final ConfigurableFileCollection classpathFiles
 
     @InputFiles
     final ConfigurableFileCollection sources
 
+    @InputFiles
+    final ConfigurableFileCollection mainClassesDirs
+
     @OutputFile
     final RegularFileProperty destination
 
     @Internal
-    TaskProvider<Copy> processResourcesTask
+    final DirectoryProperty processResourcesDir
 
     @Internal
     final Property<ProjectLayout> layout
 
-    @Internal
-    final NamedDomainObjectSet<SourceSet> sourceSets
+    private final ProviderFactory providerFactory
 
     @Inject
-    JandexTask(WorkerExecutor workerExecutor, ObjectFactory objects) {
+    JandexTask(WorkerExecutor workerExecutor, ObjectFactory objects, ProviderFactory providerFactory) {
+        this.workerExecutor = workerExecutor
+        this.providerFactory = providerFactory
         layout = objects.property(ProjectLayout)
-        sourceSets = objects.namedDomainObjectSet(SourceSet)
 
-        this.workerExecutor = workerExecutor
-        processDefaultFileSet = SimpleBooleanState.of(this, 'jandex.process.default.file.set', true)
-        includeInJar = SimpleBooleanState.of(this, 'jandex.include.in.jar', true)
-        indexName = SimpleStringState.of(this, 'jandex.index.name', 'jandex.idx')
-        indexVersion = SimpleIntegerState.of(this, 'jandex.index.version')
+        processDefaultFileSet = objects.property(Boolean)
+        includeInJar = objects.property(Boolean)
+        indexName = objects.property(String)
+        indexVersion = objects.property(Integer)
+
+        // Set final value by chaining providers with a defined precedence
+        // 1. Command-line option (set via @Option)
+        // 2. System property
+        // 3. Environment variable
+        // 4. Convention (default)
+        processDefaultFileSet.convention(providerFactory.systemProperty('jandex.process.default.file.set')
```

Comment:
Just to be clear, these providers (sysprops and envvars) are compatible with configuration cache?